cirq/google/engine/engine.py

```diff
@@ -388,6 +395,31 @@ def _infer_gcs_results(self, job_config: JobConfig) -> None:
             job_config.gcs_results = '{}jobs/{}'.format(job_config.gcs_prefix,
                                                         job_config.job_id)
 
+    def _make_request(self, request: HttpRequest) -> Dict:
+        RETRYABLE_ERROR_CODES = [500, 503]
+        current_delay = 0.1  #100ms
+
+        while True:
+            try:
+                return request.execute()
+            except ConnectionResetError:
+                message = "Lost connection to the engine."
+            except HttpError as rawErr:
+                err = json.loads(rawErr.content).get('error')
+                message = err.get('message')
+                # Raise RuntimeError for exceptions that are not retryable.
+                # Otherwise, pass through to retry.
+                if not err.get('code') in RETRYABLE_ERROR_CODES:
+                    raise RuntimeError(message) from None
+
+            current_delay *= 2
+            if current_delay > self.max_retry_delay:
+                raise Exception('Reached max retry attempts for error: ' +
+                                message)
+            print(message)
```

Comment:
Where is this being printed? This method may have been called inside a script, it's not appropriate to print like this.

At the very least the output should go to stderr instead of stdout, but I would also add a parameter to control whether or not any printed progress is shown at all.